dto/src/main/java/ca/on/oicr/gsi/runscanner/dto/Analysis.java

```diff
@@ -0,0 +1,16 @@
+package ca.on.oicr.gsi.runscanner.dto;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public abstract class Analysis<T extends WorkflowAnalysis> {
```

Comment:
@JsonTypeInfo

Reply:
@djcooke I did this, and it writes out the property name i gave it, but only that, any ideas?
![image](https://github.com/user-attachments/assets/8fb9a806-52c6-4161-9331-afd4a8d8caea)

Reply:
Update as of [f72bb63](https://github.com/miso-lims/runscanner/pull/165/commits/f72bb632cb7e5e976c786fd14792f1c43218b28a) there's also the attempt number
![image](https://github.com/user-attachments/assets/d4d92f01-4931-4db0-92cd-7905bc8d170d)

Reply:
The SampleSheetSection and its implementation might also need Jackson annotations?

Reply:
Jackson might also need explicit getters - `Analysis.analyses` and `DragenAnalysis.samplesheet` don't yet have getters

Reply:
Writing unit tests would be the easiest way to troubleshoot and much faster than scanning a run directory to test every change. Start with the smallest unit and work your way out until a full DragenAnalysis is serializing and deserializing fine. Cardea has similar tests: https://github.com/oicr-gsi/cardea/blob/main/cardea-data/src/test/java/ca/on/oicr/gsi/cardea/JacksonTest.java

Reply:
Thanks Heather, this at least got the samplesheet writing out. 
So if everything needs a getter for serialization, I imagine it also needs a setter for deserialization? How do you keep someone from modifying something they shouldn't after the fact then? For example, the samplesheet shouldn't be able to be overwritten once it's been parsed in.

Reply:
It looks like Jackson uses reflection to access private and protected properties. I've seen (but not verified) one article which claims Jackson will use reflection to populate a field if it has a getter. Writing the unit tests like Dillan suggested above will be a quick way to explore this!

Reply:
Immutability would prevent modifying the object, but Run Scanner's DTOs are not immutable

Reply:
as of https://github.com/miso-lims/runscanner/pull/165/commits/cb14227432607ce152046722836f1ee1ba9c2463: Getting this error on the DragenWorkflowAnalysis serialization test:
`com.fasterxml.jackson.databind.exc.MismatchedInputException: Cannot construct instance of `ca.on.oicr.gsi.runscanner.dto.dragen.DragenWorkflowAnalysis` (although at least one Creator exists): cannot deserialize from Object value (no delegate- or property-based Creator)`

Not sure why, because WorkflowAnalysis has the correct JsonTypeInfo annotation?

Reply:
It might actually be fine if you rename the parameter to `workflowName` so that it matches the field name it's supposed to be used for

Reply:
Unfortunately, renaming it didn't help

Reply:
Does there need to be? The annotation alone makes it serialize with both fields set correctly:

`{"suite":"DRAGEN","completionTime":31556889864403199.999999999,"startTime":-31557014167219200.000000000,"workflowName":"TEST","analyses":[{"sample":"TEST_SAMPLE","lane":1,"index":"AAAAAA-TTTTTT","files":[{"path":"file:///path/to/file","crc32Checksum":"abcdefg","size":1000,"createdTime":0.0,"modifiedTime":0.0,"info":{}}]}],"suite":"DRAGEN"}`

I would think the annotation would tell jackson to look at 'suite' at the correct level and construct the class indicated in the @Type annotation?

Reply:
Why is the constructor parameter called `name` when the WorkflowAnalysis field it maps to is `workflowName`?

Reply:
Finished all the unit tests. All passing. Still doesn't actually work when running runscanner